### PR TITLE
feat(retros-in-disguise): Base activity library + stubbed cards

### PR DESCRIPTION
**Pull request by jmtaber129**
# Description
Create the basic layout for the activity library + stubbed activity "cards".

From here, we can implement activity card designs in parallel with categories + search (which effectively just filter the activity cards shown).

Fixes https://github.com/ParabolInc/parabol/issues/7832

## Demo
Screen Shot 2023-03-02 at 4 39 26 PM

## Testing scenarios
- [ ] Confirm that team, org, and public retro + poker templates appear, and that Standup and Check-in cards are present.

## Final checklist

- [X] I checked the [code review guidelines](../docs/codeReview.md)
- [X] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [X] I have performed a self-review of my code, the same way I'd do it for any other team member
- [X] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [X] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [X] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [X] PR title is human readable and could be used in changelog

### CODE REVIEW

File: packages/client/components/ActivityLibrary/ActivityLibrary.tsx
Comment (BartoszJarocki) [inline packages/client/components/ActivityLibrary/ActivityLibrary.tsx:54]: +1 to have a better typesafety in this case it'd be better to specify type in the variable declaration
```
const templates: {
    id: string
    name: string
    type: MeetingTypeEnum
    team: {name: string}
  }[] = ...
```

as what casting does is basically saying, `not gonna check anything, I'll assume the structure is correct`

File: packages/server/graphql/public/types/User.ts
Comment (BartoszJarocki) [inline packages/server/graphql/public/types/User.ts:82]: -1 what does `0.9` mean here and why this particular value? it'd be great to either extract a const with some comment so it's easier to understand what's happening here

File: packages/server/graphql/public/types/User.ts
Comment (BartoszJarocki) [inline packages/server/graphql/public/types/User.ts:95]: -1 same here

**Review by BartoszJarocki — CHANGES_REQUESTED** [review]
Looks good! Left a few comments, it's ready to go afterwards 👍

File: packages/client/components/ActivityLibrary/ActivityLibrary.tsx
Comment (BartoszJarocki) [inline packages/client/components/ActivityLibrary/ActivityLibrary.tsx:24]: +1 github lost my comment, or I somehow deleted it. it's a little bit of overkill to have a connection  here, as we won't have too many, no harm though

File: packages/server/graphql/public/types/User.ts
Comment (BartoszJarocki) [inline packages/server/graphql/public/types/User.ts:68]: +1 is it possible to combine type definition via SDL and the old way? if yes, we should try to use the new way when we can

**Review by BartoszJarocki — APPROVED** [review]
lgtm! moving to maintainer review, @mattkrick do you mind taking a look?

File: packages/server/graphql/public/types/User.ts
Comment (jmtaber129) [reply packages/server/graphql/public/types/User.ts:68]: Could you elaborate? This is on the new SDL `User` type.

Comment (BartoszJarocki) [reply packages/server/graphql/public/types/User.ts:68]: oh god, long day, sorry 🤦

File: packages/client/components/ActivityLibrary/ActivityLibrary.tsx
Comment (jmtaber129) [reply packages/client/components/ActivityLibrary/ActivityLibrary.tsx:24]: Hmm I don't disagree, but since the existing `organizationTemplates` and `publicTemplates` fields on the meeting settings objects are connections, I have a weak preference to either a) make this a connection, too, or b) make this an array and make those fields arrays.

**Review by mattkrick — CHANGES_REQUESTED** [review]
a few minor things we can fix later, but only 1 thing blocking the merge.

File: packages/client/components/ActivityLibrary/ActivityLibraryCard.tsx
Comment (jmtaber129) [reply packages/client/components/ActivityLibrary/ActivityLibraryCard.tsx:9]: I don't foresee us using this full component in the `/meetings` dashboard, since it will eventually have onClick handlers, etc. for entering the activity details view. 

We can refactor the relevant parts of this into its own component once we add the UI portion of this to the dashboard.

File: packages/server/graphql/queries/helpers/getScoredTemplates.ts
Comment (jmtaber129) [reply packages/server/graphql/queries/helpers/getScoredTemplates.ts:4]: Interesting context! However, to limit the scope of this PR, I'd prefer to leave this as-is for now, and we can update the ranking logic at a later date.

File: packages/server/utils/getTemplateScore.ts
Comment (jmtaber129) [inline packages/server/utils/getTemplateScore.ts:20]: @mattkrick this came up when I was regression testing removing the team templates DB call - like you mentioned we'll probably change how we rank templates at some point, but I figured we should fix make this fix, since otherwise all templates will score 0 due to age being millis when the logic assumes it's days.

File: packages/server/graphql/public/types/User.ts
Comment (jmtaber129) [reply packages/server/graphql/public/types/User.ts:79]: Good catch! Removed the team templates call and pulled those templates from these results instead.

File: packages/client/components/ActivityLibrary/ActivityLibraryCard.tsx
Comment (BartoszJarocki) [reply packages/client/components/ActivityLibrary/ActivityLibraryCard.tsx:9]: will allow myself to nitpick a little

1. Parts of this component should be reusable as they're exactly the same (ie image part with this rounded thingy in the corners)
Screenshot 2023-03-09 at 07 23 11

2. imo activity details should be a route like `/activity-library/:activity-id` which preferably makes this component a link (this also will be good for accessibility and keyboard navigation reasons), see the video below

https://user-images.githubusercontent.com/1017620/223938475-b73a31c0-7f62-47b2-ba18-0221a742ac96.mp4

if it's a link we can easily use tab + enter to navigate that view, which is a win.

it's not really related to this pr though 🙃

File: packages/client/components/ActivityLibrary/ActivityLibrary.tsx
Comment (jmtaber129) [reply packages/client/components/ActivityLibrary/ActivityLibrary.tsx:51]: I'm thinking since we'll probably need to do some special-casing on the client-side for standups+check-ins, it _might_ be better to add these on the client-side - i.e. the server gives the client all the meeting _templates_, and it's up to the client to use those templates to show the available activities.

File: packages/client/components/ActivityLibrary/ActivityLibrary.tsx
Comment (jmtaber129) [reply packages/client/components/ActivityLibrary/ActivityLibrary.tsx:24]: I kind of disagree with this for the reasons mentions below - we don't have an "activity" object in the server-side data model, but we do have templates + meetings, so the server should return templates, and then the client uses those templates to display available activities.

File: packages/server/utils/getTemplateScore.ts
Comment (mattkrick) [reply packages/server/utils/getTemplateScore.ts:20]: sounds good!

**Pull request merged**
